<!-- 
    ROS navigation stack with velocity smoother and safety (reactive) controller
-->
<launch>
  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="scan" />
  <arg name="cmd_vel_topic" default="/mobile_base/commands/velocity" />
  <arg name="namespace" default= "" />

  <node pkg="move_base" type="move_base" respawn="true" name="move_base">
    <rosparam file="$(find distributed_slam_karto)/launch/includes/costmap_common_params.yaml" command="load" ns="global_costmap"/> 
    <rosparam file="$(find distributed_slam_karto)/launch/includes/global_costmap_params.yaml" command="load" />

  <rosparam file="$(find distributed_slam_karto)/launch/includes/costmap_common_params.yaml" command="load" ns="local_costmap"/>
  <rosparam file="$(find distributed_slam_karto)/launch/includes/local_costmap_params.yaml" command="load" />


    <rosparam file="$(find distributed_slam_karto)/launch/includes/global_planner_params.yaml" command="load" />

    <rosparam file="$(find distributed_slam_karto)/launch/includes/move_base_params.yaml" command="load" />
    <!-- external params file that could be loaded into the move_sbase namespace -->
    <rosparam file="$(find distributed_slam_karto)/launch/includes/pf_local_planner.yaml" command="load" />
    <!-- reset frame_id parameters using user input data -->

    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="global_costmap/static/map_topic" value="/$(arg namespace)/map"/>

    <param name="local_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>

    <param name="PFLocalPlanner/goal_frame_id" value="$(arg global_frame_id)" />
    <param name="PFLocalPlanner/cmd_frame_id" value="$(arg base_frame_id)" />
    <param name="PFLocalPlanner/scan_topic" value="$(arg laser_topic)" />
    <param name="PFLocalPlanner/status_topic" value="/$(arg namespace)/pf_status" />

    <remap from="base_scan" to="$(arg namespace)/scan"/>
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
  </node>
</launch>
